File: Hydroq/src/GameEndDialog.cpp
#include "GameEndDialog.h"

void GameEndDialog::OnInit() {
	SubscribeForMessages(ACT_BUTTON_CLICKED);
	model = GETCOMPONENT(PlayerModel);
}

void GameEndDialog::OnMessage(Msg& msg) {
	if (msg.HasAction(ACT_BUTTON_CLICKED)) {
		auto sceneContext = GETCOMPONENT(Stage);
		if (msg.GetContextNode()->GetTag().compare("ok_but") == 0) {
			sceneContext->SwitchBackToScene(TweenDirection::NONE);
			sceneContext->SwitchBackToScene(TweenDirection::NONE);
		}
	}
}

void GameEndDialog::Update(const uint64 delta, const uint64 absolute) {
	if (!firstInit && model->GameEnded()) {
		// stop all sounds
		auto sounds = CogGetPlayedSounds();
		for (auto sound : sounds) {
			sound->Stop();
		}

		auto msgNode = owner->GetScene()->FindNodeByTag("gameend_msg");
		auto player = GETCOMPONENT(PlayerModel);

		msgNode->GetMesh<Text>()->SetText(!player->PlayerWin() ? "You lost the game!" : "You win the game!");
		firstInit = true;
	}
}

File: Hydroq/data/scripts/GameEndDialog.lua
 GameEndDialog = { 
   firstInit = false,
   model = nil,
   OnInit = function() 
     GameEndDialog_beh:SubscribeForMessages(StrId("BUTTON_CLICKED"))
	 model = HydroqGetPlayerModel()
   end, 
   OnMessage = function(msg) 
     if(msg:HasAction(StrId("BUTTON_CLICKED")))
	 then
	   if(msg.contextNode.tag == "ok_but")
	   then
	     CogSwitchBackToScene("none")
		 CogSwitchBackToScene("none")
	   end
	 end
   end, 
   Update = function(delta, absolute)
     if(not firstInit and model:GameEnded())
	 then
	   CogStopAllSounds();
	   scene = GameEndDialog_beh:GetScene()
	   msgNode = scene:FindNodeByTag("gameend_msg")
	   playerWin = model:PlayerWin()
	   
	   meshText = msgNode:GetTextMesh()
	   if(playerWin)
	   then
	     meshText:SetText("You have won the game!")
	   else
	     meshText:SetText("You have lost the game!")
	   end
	   
	 end
   end 
} 

GameEndDialog_beh = Behavior() 
GameEndDialog_beh:Register(GameEndDialog, "GameEndDialog")